src/App.js

```diff
@@ -1,26 +1,205 @@
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import React from "react";
+import ReactDOM from "react-dom";
+import "./App.css";
+import * as d3 from "d3";
+import Norway from "./Norway";
+
+class ElectionTypes extends React.Component {
+  state = {
+    election_types: []
+  };
+
+  handleError(res, message) {
+    console.log("ERROR (" + res.status + "): " + message);
+  }
+
+  getElectionTypes() {
+    fetch("https://sleepy-retreat-45150.herokuapp.com/api/election_types")
+      .then(response => {
+        if (!response.ok) {
+          this.handleError(response, "Failed to get election types");
+        }
+        return response.json();
+      })
+      .then(data => {
+        this.setState({ election_types: data.election_types });
+      });
+  }
+
+  componentDidMount() {
+    this.getElectionTypes();
+  }
+
+  render() {
+    return (
+      <ul>
+        {this.state.election_types.map(function(item) {
+          return <li key={item.id}>{item.name}</li>;
+        })}
+      </ul>
+    );
+  }
+}
+
+class ElectionEvents extends React.Component {
+  state = {
+    election_events: []
+  };
+
+  handleError(res, message) {
+    console.log("ERROR (" + res.status + "): " + message);
+  }
+
+  getElectionEvents() {
+    fetch("https://sleepy-retreat-45150.herokuapp.com/api/election_events")
+      .then(response => {
+        if (!response.ok) {
+          this.handleError(response, "Failed to get election events");
+        }
+        return response.json();
+      })
+      .then(data => {
+        this.setState({ election_events: data.election_events });
+      });
+  }
+
+  componentDidMount() {
+    this.getElectionEvents();
+  }
+
+  render() {
+    return (
+      <ul>
+        {this.state.election_events.map(function(item) {
+          return <li key={item.id}>{item.name}</li>;
+        })}
+      </ul>
+    );
+  }
+}
+
+class Elections extends React.Component {
+  state = {
+    elections: []
+  };
+
+  handleError(res, message) {
+    console.log("ERROR (" + res.status + "): " + message);
+  }
+
+  getElections() {
+    fetch("https://sleepy-retreat-45150.herokuapp.com/api/elections")
+      .then(response => {
+        if (!response.ok) {
+          this.handleError(response, "Failed to get elections");
+        }
+        return response.json();
+      })
+      .then(data => {
+        this.setState({ elections: data.elections });
+      });
+  }
+
+  componentDidMount() {
+    this.getElections();
+  }
+
+  render() {
+    return (
+      <ul>
+        {this.state.elections.map(function(item) {
+          return (
+            <li key={item.id}>
+              {item.election_event} {item.election_type}
+            </li>
+          );
+        })}
+      </ul>
+    );
+  }
+}
+
+class BarChart extends React.Component {
+  componentDidMount() {
+    this.svg = ReactDOM.findDOMNode(this);
```

Comment:
It's generally preferrable to use `ref`s for this sort of thing: https://reactjs.org/docs/refs-and-the-dom.html

So it's like:
```
this.svg = React.createRef();
```
then in render:

```
  render(){
    return <svg width={this.props.width} height={this.props.height} ref={this.svg} />
  }
```

Then in componentDidMount or componentDidUpdate, you'd fetch the ref:

```
componentDidUpdate() {
  this.drawChart(this.svg.current)
}
```

Now this will still be potentially slow, as you'd need to implement shouldComponentUpdate and a bunch of other things — as everything beneath this component (all the child that exist of <svg>) aren't handled by React. That led to some pretty awkward code, but you can now use [Hooks](https://reactjs.org/docs/hooks-intro.html) to clean this up a fair bit: https://medium.com/stationfive/how-to-create-a-pie-chart-with-d3-js-and-react-hooks-part-1-81bcd7f39b32